### ISSUE TITLE
SetupWizard > On My Own -- Handle username and device name length validation errors

### ISSUE THREAD

**Issue by nucleogenesis** [learningequality/kolibri#10326]
## Observed behavior

During the "On my own" flow, the user's device name is automatically generated to something like "Personal Device for <the users' full name>". 

An issue arises wherein the `device name` can only be up to 50 characters long -- but the user's `full_name` can be up to 120 which can easily result in the device name being too long during provisioning.

## Expected behavior

_Idea 1: Interpolation_

@rtibbles suggestion (if I understand correctly) - is that we would actually store the user's FacilityUser ID in the string and then use JS on the front-end to interpolate the user's full name in the ID's place when rendering the string. This is a step in a complex direction and, considering the UUID is 32 characters long, that only leaves 18 characters for the rest of the translated strings which could still lead to an issue.

One concern with the idea of _increasing the validation's limit of 50 characters_ on the device name is that we may run into issues with ZeroConf / discovery stuff -- but that concern was discussed in the context of making it 170 characters (or whatever -- to accommodate the longest possible) -- _however_ - we may find that increasing it by 20-30 characters could accommodate the solution above.

_Idea 2: Truncation_

Rather than handling this on the backend at all, we may instead opt to just save a truncated version of the user's full_name, if needed, to set the device name to the "Personal device for ..." such that it will only ever be 50 characters long. Something like:

```
let baseTranslatedLength = this.$tr('defaultDeviceName', { name: '' }).length
let truncatedFullName = this.user.full_name.slice(0, baseTranslatedLength-1);
let actualStringWeUse = this.$tr('defaultDeviceName', { name: truncatedFullName });
```

should do it and take into account translated message links.

---

_Idea 3: ???_

This one can get a bit hairy, so any additional thoughts from @bjester @rtibbles would  be much appreciated here.

## User-facing consequences

Cannot provision a device if you have a long full name or speak German and have a medium full name.

**Comment by holta** [comment]
Coincidentally I had the exact same question yesterday: what validation of KOLIBRI_USER is wise and/or necessary?!

But not knowing the exact answer (specifically in the case of Linux here) I just validated the input file to be non-empty for now:

```
if [ -s /etc/kolibri/username ]    # Test that file exists with non-zero size
```

@jredrejo might have better ideas!  See line 349 here, in what is basically a 2-line fix to Kolibri's Debian/Ubuntu installer:

- PR learningequality/kolibri-installer-debian#117

**Comment by rtibbles** [comment]
@holta this issue is just related to usernames within Kolibri, not system usernames.

**Comment by rtibbles** [comment]
I think for simplicity we should just do this with Idea 2. These facilities are only for the 'on my own' workflow, so are not intended to be massively meaningful in their name. If really needed, the facility can be renamed.

**Comment by rtibbles** [comment]
Fixed in #10392
